source/Octopus.Server.Client/OctopusServerEndpoint.cs

```diff
@@ -8,9 +8,31 @@ namespace Octopus.Client
     /// </summary>
     public class OctopusServerEndpoint
     {
-        readonly ILinkResolver octopusServer;
-        readonly string apiKey;
-        readonly ICredentials credentials;
+        /// <summary>
+        /// Create an instance with a Token to authenticate.
+        /// </summary>
+        public static OctopusServerEndpoint CreateWithToken(string octopusServerAddress, string token, ICredentials credentials = null)
+        {
+            return new OctopusServerEndpoint(octopusServerAddress, new TokenValue(token), credentials);
+        }
```

Comment:
I couldn't find a particularly "nice" way to add the ability to specify a token without changing the existing public constructors in some way. My initial idea was to add an optional `bool isApiKey = false` parameter to the end of the appropriate constructor but that felt a bit too ugly. I decided to go for some public factory methods which are explicit and then have a private constructor taking a Token so it's not part of the public API.

Reply:
The main reason this is an issue is because the Token and the ApiKey are both strings, so adding a new constructor for the token would clash with the existing constructor/s which take API keys.